src/main/scala/monix/kafka/KafkaConsumer.scala

```diff
@@ -0,0 +1,140 @@
+package monix.kafka
+
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Ack.Stop
+import monix.execution.FutureUtils.extensions.FutureExtensions
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.{Observable, Observer}
+import monix.reactive.observers.Subscriber
+import org.apache.kafka.clients.consumer.{ConsumerRecord, ConsumerRecords, KafkaConsumer => ApacheKafkaConsumer}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+import scala.collection.JavaConverters._
+
+/** An `Observable` implementation that reads messages from `Kafka`.
+  *
+  * You need to call `addTopicSubscription` in order for the Observable to start sending messages from that topic.
+  */
+final class KafkaConsumer[K, V] private
+  (config: KafkaConsumerConfig, ioScheduler: Scheduler)
+  (implicit K: Deserializer[K], V: Deserializer[V])
+  extends Observable[ConsumerRecord[K, V]] with StrictLogging {
+
+  def unsafeSubscribeFn(subscriber: Subscriber[ConsumerRecord[K, V]]): Cancelable = {
+    val source = underlying.onErrorHandleWith { ex =>
+      logger.error(
+        s"Unexpected error in KafkaConsumerObservable, retrying in ${config.retryBackoffTime}", ex)
+      // Applying configured delay before retry!
+      underlying.delaySubscription(config.retryBackoffTime)
+    }
+
+    source.unsafeSubscribeFn(subscriber)
+  }
+
+  def addTopicSubscription(topics: Set[String]): Unit = {
+    logger.info(s"Subscribing to topics : ${topics.mkString(",")}")
+    consumerRef.subscribe((currentSubscriptions() ++ topics).asJavaCollection)
+  }
+
+  /**
+    * Removes a subscription from a particular topic.
+    *
+    * @param topics the topics to be unsubscribed, if it is an `Set.empty` this method will unsubscribe from all topics.
+    */
+  def removeTopicSubscription(topics: Set[String]): Unit =
+    if(topics.nonEmpty) {
+      logger.info(s"Unsubscribing from topics : ${topics.mkString(",")}")
+      consumerRef.subscribe((currentSubscriptions() -- topics).asJavaCollection)
+    } else {
+      logger.info(s"Unsubscribing from topics : ${currentSubscriptions().mkString(",")}")
+      consumerRef.unsubscribe()
+    }
+
+  // Gets initialized on `subscribe`
+  private lazy val consumerRef = {
```

Comment:
This consumer reference cannot be a member of the class. Because this observable is cold, which means a `consumerRef` should be created on each `unsafeSubscribeFn` call.